task-annotation-generator.js

```diff
@@ -4,11 +4,43 @@
 
 var fs = require('fs');
 
-function TaskAnnotation(validator) {
-    this.validator = validator;
+function TaskAnnotation(task) {
+    this.task = task;
 }
 
 /**
+ * Run the process to generate doc data with current existing task schemas
+ * @param {Array} tasks - list of task from task folder
+ *                        e.g. /lib/task-data/tasks/*.js
+ * @param {Object} validator - instance of task-option-validator
+ * @return {Promise} promise with value docData array
+ */
+TaskAnnotation.run = function (tasks, validator) {
+    return validator.register()
+    .return(tasks)
+    .filter(function (task) {
+        // filter all tasks, only those task has schema will be annotated.
+        return task.schemaRef ? true: false;
+    })
+    .map(function (task) {
+        var tn = new TaskAnnotation(task);
+        var schemaResolved = validator.getSchemaResolved(task.schemaRef);
+        var schemaMerged = tn.mergeSchema(schemaResolved);
+        tn.addDefault(schemaMerged);
+        return tn.generateDocData(schemaMerged, {
+            // url: '/' + task.injectableName,
+            name: 'option',
+            title: 'option',
+            group: task.injectableName.replace(/\./g, '_'),
```

Comment:
The `injectableName` here would finally be using as an anchor in HTML.
For example, 
```HTML
<a href="#api-Task.Os.Install.Ubuntu" />
``` 
The anchor originally has scrolling animation, but it broke when I use the anchor link as above.
It works when I change to 
```HTML
<a href="#api-Task_Os_Install_Ubuntu" />
```
But I didn't dig into the root cause